src/Common/SettingsChanges.h

```diff
@@ -6,6 +6,14 @@
 namespace DB
 {
 
+enum class SettingChangeSource
```

Comment:
Please add some SQL examples as comments to each "setting changes source", otherwise the enum values are too confusing.

Reply:
Done!